### PR TITLE
Version 2: Add certificate creation logic

### PR DESCRIPTION
**Pull request by OleksandrSPV**
https://github.com/PeculiarVentures/fortify-tools/assets/129727832/df52cf01-2c26-4b22-afdb-c0b7dd09b18e

### CODE REVIEW

File: src/hooks/app/useApp.tsx [hunk 1]
```diff
@@ -199,6 +199,52 @@ export function useApp() {
...
+        await fortifyClient.current.getCertificatesByProviderId(providerId)
+      );
+      setCurrentProviderId(providerId);
+      setFetchingValue("certificates", "resolved");
+    } catch (error) {
+      setFetchingValue("certificates", "rejected");
+    }
+  };
+
+  const handleAddCSRToList = (
```
Comment (donskov) [inline src/hooks/app/useApp.tsx:220]: We don't need this extra code. Let's reload the list of certificates and in feature, we will add CSR support.

File: src/dialogs/certificate-create-dialog/useCertificateCreateDialog.tsx [hunk 2]
```diff
@@ -13,8 +18,15 @@ import { CertificateType } from "../../types";
 export function useCertificateCreateDialog(props: {
   providers: IProviderInfo[];
   currentProviderId?: string;
+  fortifyClient: FortifyAPI | null;
```
Comment (donskov) [inline src/dialogs/certificate-create-dialog/useCertificateCreateDialog.tsx:21]: The hook can't work if the `fortifyClient` prop is not defined.

**Review by donskov — CHANGES_REQUESTED** [review]

File: src/hooks/app/useApp.tsx (hunk 1 above)
Comment (OleksandrSPV) [reply src/hooks/app/useApp.tsx:220]: Ok, deal

Comment (OleksandrSPV) [reply src/hooks/app/useApp.tsx]: Removed here https://github.com/PeculiarVentures/fortify-tools/pull/149/commits/ba3030083b7ffd242494e4077bba9baa386c5203

File: src/dialogs/certificate-create-dialog/useCertificateCreateDialog.tsx (hunk 2 above)
Comment (OleksandrSPV) [reply src/dialogs/certificate-create-dialog/useCertificateCreateDialog.tsx]: Fixed here https://github.com/PeculiarVentures/fortify-tools/pull/149/commits/ba3030083b7ffd242494e4077bba9baa386c5203

File: src/dialogs/certificate-create-dialog/useCertificateCreateDialog.tsx [hunk 3]
```diff
@@ -13,39 +18,80 @@ import { CertificateType } from "../../types";
...
-    setTimeout(function () {
-      setIsLoading(false);
+    try {
+      let newCert;
+      if (type === "x509") {
+        newCert = await fortifyClient.createX509(
+          localCurrentProviderId.current as string,
+          {
+            subjectName: subject,
+            hashAlgorithm: algorithm?.hash as EHashAlgorithm,
```
Comment (donskov) [inline src/dialogs/certificate-create-dialog/useCertificateCreateDialog.tsx:55]: I don't see any gap to make these variables required and prevent using `as`. Any thoughts?

**Review by donskov — CHANGES_REQUESTED** [review]

File: src/dialogs/certificate-create-dialog/useCertificateCreateDialog.tsx (hunk 3 above)
Comment (OleksandrSPV) [reply src/dialogs/certificate-create-dialog/useCertificateCreateDialog.tsx]: @donskov Fixed here https://github.com/PeculiarVentures/fortify-tools/pull/149/commits/604e624eb1a7d4d0ab20e005e8a5065b72656c04

Comment (OleksandrSPV) [reply src/dialogs/certificate-create-dialog/useCertificateCreateDialog.tsx]: @donskov Fixed here https://github.com/PeculiarVentures/fortify-tools/pull/149/commits/604e624eb1a7d4d0ab20e005e8a5065b72656c04

**Review by donskov — APPROVED** [review]

**Pull request merged**